### PR TITLE
getex

### PR DESCRIPTION
**Pull request by AvitalFineRedis**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [X] Does `$ tox` pass with this change (including linting)?
- [X] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [X] Is the new or changed code fully tested?
- [X] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

Add support to GETEX

### CODE REVIEW

**Comment by AvitalFineRedis** [comment]
@chayim

File: redis/client.py [hunk 1]
```diff
@@ -1689,6 +1689,54 @@ def get(self, name):
...
         return self.execute_command('GET', name)
 
+    def getex(self, name,
+              ex=None, px=None, exat=None, pxat=None, persist=False):
+        """
+        Get the value of key and optionally set its expiration.
+        GETEX is similar to GET, but is a write command with
+        additional options.
+
+        ``ex`` Set the specified expire time, in seconds.
```
Comment (chayim) [inline redis/client.py:1699]: Can the comment format change to match others in this class?

File: redis/client.py [hunk 2]
```diff
@@ -1689,6 +1689,54 @@ def get(self, name):
...
+        will expire, in seconds.
+
+        ``pxat`` Set the specified Unix time at which the key
+        will expire, in milliseconds.
+
+        ``persist`` Remove the time to live associated with the key.
+        """
+
+        pieces = []
+        if ex is not None:
```
Comment (chayim) [inline redis/client.py:1713]: ex should be an integer that is > 0 instead of a datetime.timedelta

File: redis/client.py [hunk 3]
```diff
@@ -1689,6 +1689,54 @@ def get(self, name):
...
+            pieces.append('EX')
+            if isinstance(ex, datetime.timedelta):
+                ex = int(ex.total_seconds())
+            pieces.append(ex)
+        if px is not None:
+            pieces.append('PX')
+            if isinstance(px, datetime.timedelta):
```
Comment (chayim) [inline redis/client.py:1720]: px should be an int since we convert anyways.

File: redis/client.py [hunk 4]
```diff
@@ -1689,6 +1689,54 @@ def get(self, name):
...
+    def getex(self, name,
+              ex=None, px=None, exat=None, pxat=None, persist=False):
+        """
+        Get the value of key and optionally set its expiration.
+        GETEX is similar to GET, but is a write command with
+        additional options.
+
+        ``ex`` Set the specified expire time, in seconds.
+
+        ``px`` Set the specified expire time, in milliseconds.
```
Comment (chayim) [inline redis/client.py:1701]: Let's specify the types in the comment

File: tests/test_commands.py [hunk 5]
```diff
@@ -727,6 +727,21 @@ def test_get_and_set(self, r):
         assert r.get('integer') == str(integer).encode()
         assert r.get('unicode_string').decode('utf-8') == unicode_string
 
+    @skip_if_server_version_lt('6.2.0')
+    def test_getex(self, r):
+        assert r.set('a', 1)
```
Comment (chayim) [inline tests/test_commands.py:732]: Shouldn't need to assert the sets

File: tests/test_commands.py [hunk 6]
```diff
@@ -727,6 +727,21 @@ def test_get_and_set(self, r):
...
+        assert r.getex('a') == b'1'
+        assert r.ttl('a') == -1
+        assert r.getex('a', ex=60) == b'1'
+        assert r.ttl('a') == 60
+        assert r.getex('a', px=6000) == b'1'
+        assert r.ttl('a') == 6
+        expire_at = redis_server_time(r) + datetime.timedelta(minutes=1)
+        assert r.getex('a', pxat=expire_at) == b'1'
```
Comment (chayim) [inline tests/test_commands.py:740]: Need to update the tests to match the types specified above.

File: redis/client.py [hunk 7]
```diff
@@ -1689,6 +1689,54 @@ def get(self, name):
...
+                ex = int(ex.total_seconds())
+            pieces.append(ex)
+        if px is not None:
+            pieces.append('PX')
+            if isinstance(px, datetime.timedelta):
+                px = int(px.total_seconds() * 1000)
+            pieces.append(px)
+        if exat is not None:
+            pieces.append('EXAT')
+            if isinstance(exat, datetime.datetime):
```
Comment (chayim) [inline redis/client.py:1725]: Same types comment.

File: redis/client.py [hunk 8]
```diff
@@ -1689,6 +1689,54 @@ def get(self, name):
...
+                s = int(exat.microsecond / 1000000)
+                exat = int(mod_time.mktime(exat.timetuple())) + s
+            pieces.append(exat)
+        if pxat is not None:
+            pieces.append('PXAT')
+            if isinstance(pxat, datetime.datetime):
```
Comment (chayim) [inline redis/client.py:1731]: Given the other comments, does this still make sense?

**Comment by AvitalFineRedis** [comment]
@chayim

**Comment by chayim** [comment]
Thanks for your contirbution!

**Pull request merged**